File: Python/Algorithms/2103.py
class Solution:
    def countPoints(self, rings: str) -> int:
        GREEN, RED, BLUE = 1, 2, 4
        rods = [0] * 10
        
        for i in range(1, len(rings), 2):
            color, rod = rings[i-1], rings[i]
            if color == 'R':
                b = RED
            elif color == 'B':
                b = BLUE
            else:
                b = GREEN
            rods[int(rod)] |= b
            
        # RED | BLUE | GREEN == 7
        return sum(i == 7 for i in rods)

File: Rust/Algorithms/2103.rs
impl Solution {
    pub fn count_points(rings: String) -> i32 {
        let (RED, BLUE, GREEN) = (1, 2, 4);
        let mut rods = vec![0; 10];
        
        let mut it = rings.chars();
        while let Some(color) = it.next() {
            // ok due to constraints.
            let rod = it.next().unwrap().to_digit(10).unwrap() as usize;
            rods[rod] |= match color {
                'R' => RED,
                'G' => GREEN,
                'B' => BLUE,
                _ => 0
            };
        }

        rods.iter().filter(|&&x| x == 7).count() as i32
    }
}